sagemathinc/cocalc
# webapp/latex: count words

# Description
latex word count panel

## Testing Steps
1. open a tex file and select the word count panel, you should see its output
2. similarly, this also works for knitr documents, because it still parses the generated tex file
3. to test if auto updating works:
    1. when hidden and shown again: switch the pdf preview panel to word count, remember the word count you see, switch back to pdf preview (i.e. in such a way, that the word count panel isn't shown anywhere). Then edit the document and save it. Then switch back to the word count panel and see the updated count
    2. when building: edit the tex file and save it, the word count should update.
4. clicking the button is a forced build

## Need Help
> What's the canonical way to trigger this once when the panel is shown? It's a bit weird to have it empty when someone opens it up. Alternatively, if it isn't run, there would be some help text missing telling the user to click the button above.

It's triggered when the component mounts and during the run_build step.

## Relevant Issues
* #2215 

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):
